packages/store-ui/src/molecules/Modal/useTrapFocus.ts

```diff
@@ -0,0 +1,103 @@
+import { useEffect, useRef } from 'react'
+import type { RefObject } from 'react'
+import type { FocusableElement } from 'tabbable'
+import { tabbable } from 'tabbable'
+
+interface TrapFocusParams {
+  sentinelStartRef: RefObject<HTMLElement>
+  trapFocusRef: RefObject<HTMLElement>
+  sentinelEndRef: RefObject<HTMLElement>
+}
+
+/**
+ * Element that will maintain the focus inside, focus the first element,
+ * and focus back on the element that was in focus when useTrapFocus was triggered.
```

Comment:
```suggestion
 * Element that will maintain the focus inside trapFocusRef, focus the first element,
 * and, when the focus changes and the element in focus is removed from the screen, focus back on the element that was in focus when useTrapFocus was triggered.
```

Reply:
I think your suggestion makes the sentence more difficult to read.

Reply:
ahahahahha okay then!